### PR TITLE
(QA-2358) Add CLI Interface for Version

### PR DESCRIPTION
**Pull request by cowofevil**
Add a command for printing the version of the vmpooler-client. This
required changing how commands are generated in the CommandParser.
Also, I corrected some small doc string issues.

### CODE REVIEW

File: vmpooler_client/command_parser.py [hunk 1]
```diff
@@ -65,28 +65,67 @@ def __init__(self, argv):
...
-      |KeyError| Command name has already been defined.
+      |KeyError| = Command name has already been defined.
     """
 
     if cmd_name not in self._commands:
       command = self._sub_parsers.add_parser(cmd_name, description=desc)
 
       # Add a subparser for sub-commands under a top-level command. (i.e. "list" for "vm")
-      self._commands[cmd_name] = command.add_subparsers()
+      if func is None:
```
Comment (jpinsonault) [inline vmpooler_client/command_parser.py]: Not sure I care, but maybe it'd be cleaner to just do `if func` and have the positive case first, or maybe `if not func`

File: vmpooler_client/command_parser.py [hunk 2]
```diff
@@ -65,28 +65,67 @@ def __init__(self, argv):
...
+    Returns:
+      |None|
+
+    Raises:
+      |KeyError| = The command specified does not exist. The **kwargs dictionary is
+        missing the required "name" key.
+    """
+
+    if command not in self._commands:
+      raise KeyError("The '{}' command has not been defined yet!".format(parent))
```
Comment (jpinsonault) [inline vmpooler_client/command_parser.py]: Doesn't look like `parent` is defined at this point

File: vmpooler_client/command_parser.py [hunk 3]
```diff
@@ -65,28 +65,67 @@ def __init__(self, argv):
...
+        self._commands[cmd_name] = command.add_subparsers()
+
+        # Tack on this bit of meta-data to indicate that sub-commands are ALLOWED
+        self._commands[cmd_name]._allow_sub_cmd = True
+      else:
+        self._commands[cmd_name] = command
+        self._commands[cmd_name].set_defaults(func=func)
+
+        # Tack on this bit of meta-data to indicate that sub-commands are NOT allowed.
+        self._commands[cmd_name]._allow_sub_cmd = False
```
Comment (jpinsonault) [inline vmpooler_client/command_parser.py]: This is mildly gross, but the user facing API is nice, and I don't have any better ideas at the moment

File: vmpooler_client_app.py [hunk 4]
```diff
@@ -11,10 +11,12 @@
 #===================================================================================================
 # Imports
 #===================================================================================================
+from __future__ import print_function
```
Comment (jpinsonault) [inline vmpooler_client_app.py]: \o/

**Comment by jpinsonault** [comment]
Just a nitpick and an undefined variable, otherwise I'm :+1:

File: vmpooler_client/command_parser.py (hunk 3 above)
Comment (cowofevil) [inline vmpooler_client/command_parser.py]: I have a different strategy for doing this as well. I'll make the change and we can decide if it makes more sense.

File: vmpooler_client_app.py (hunk 4 above)
Comment (cowofevil) [inline vmpooler_client_app.py]: THE FUTURE IS NOW!

**Comment by cowofevil** [comment]
@jpinsonault Ping.

File: vmpooler_client/command_parser.py [hunk 5]
```diff
@@ -110,6 +146,8 @@ def add_sub_command(self, parent, sub_cmd_name, desc, func):
 
     if parent not in self._commands:
       raise KeyError("The '{}' parent command has not been defined yet!".format(parent))
+    elif isinstance(self._commands[parent], argparse.ArgumentParser):
```
Comment (jpinsonault) [inline vmpooler_client/command_parser.py]: Ooh, tricky. Do you know what class it'll be if it's not a `argparse.ArgumentParser`? Might be worth putting a comment in explaining it for future generations

Comment (cowofevil) [inline vmpooler_client/command_parser.py]: Yeah, it would be ```_SubParsersAction``` otherwise. I'll update with a comment.

**Comment by cowofevil** [comment]
@jpinsonault Updated with comment.

**Comment by jpinsonault** [comment]
Sweet. :+1:

**Pull request merged**